evalml/model_understanding/force_plots.py

```diff
@@ -0,0 +1,100 @@
+            Defaults to False.
+
+    Returns:
+        list(dict(shap.AdditiveForceVisualizer)): The same as force_plot(), but with an additional
+            key in each dictionary for the plot.
+    """
+
+    def gen_force_plot(shap_values, training_data, expected_value, matplotlib):
+        """ Helper function to generate a single force plot. """
+        # Ensure there are as many features as shap values.
+        assert training_data.shape[1] == len(shap_values)
+
+        # TODO: Update this to make the force plot display multi-row array visualizer.
+        training_data_sample = training_data.iloc[0]
+        shap_plot = shap.force_plot(expected_value, np.array(shap_values), training_data_sample, matplotlib=matplotlib)
+        return shap_plot
+
+    shap_plots = force_plot(pipeline, rows_to_explain, training_data, y)
+    for row in shap_plots:
+        for cls in row:
+            cls_dict = row[cls]
+            cls_dict["plot"] = gen_force_plot(cls_dict["shap_values"],
+                                              training_data[cls_dict["feature_names"]],
+                                              cls_dict["expected_value"],
+                                              matplotlib=matplotlib)
+    return shap_plots
+
+
+def force_plot(pipeline, rows_to_explain, training_data, y):
+    """ Function to generate the data required to build a force plot.
+
+    Args:
+        pipeline (PipelineBase): The pipeline to generate the force plot for.
+        rows_to_explain (list(int)): A list of the indices of the training_data to explain.
+        training_data (pandas.DataFrame): The data used to train the pipeline.
+        y (pandas.Series): The target data.
+
+    Returns:
+        list(dict()): list of dictionaries where each dict contains force plot data.  Each dictionary
+            entry represents the explanations for a single row.
+            e.x. For single row binary force plots:
+                    [{'malignant': {'expected_value': 0.37,
+                                    'feature_names': ['worst concave points', 'worst perimeter', 'worst radius'],
+                                    'shap_values': [0.09, 0.09, 0.08],
+                                    'plot': AdditiveForceVisualizer}]
+                 For two row binary force plots:
+                    [{'malignant': {'expected_value': 0.37,
+                                    'feature_names': ['worst concave points', 'worst perimeter', 'worst radius'],
+                                    'shap_values': [0.09, 0.09, 0.08],
+                                    'plot': AdditiveForceVisualizer},
+                    {'malignant': {'expected_value': 0.29,
+                                    'feature_names': ['worst concave points', 'worst perimeter', 'worst radius'],
+                                    'shap_values': [0.05, 0.03, 0.02],
+                                    'plot': AdditiveForceVisualizer}]
```

Comment:
Is the value of this `'plot'` key a class or a string? I ask because if its a class, we can't easily JSON-serialize this output, so we'd likely have to convert to string somewhere in order for a web UI to consume this.

Reply:
I didn't think the Web UI was going to consume this.  My understanding was that the graphing function was going to be for the purpose of EvalML users via jupyter notebook, etc. and that the data function, which didn't add the plot to the dict was going to be the one driving the web UI (which was going to render the force plots itself to get all the nice javascript stuff that shap does)

Reply:
Agreed. But isn't `force_plot` (aka this code) the data function and `graph_force_plot` the graphing function?